wpengine/faustjs
# Fix drafts for unpublished posts

## Tasks


## Description

These changes fix a bug where Faust's seed query was failing when attempting to view an unpublished WordPress post w/ a draft status.

## Testing

1. Checkout this branch: `MERL-931-preview-for-draft`
2. Run `npm run build`
3. Go to `examples/faustwp-getting-started/.env.local` and update NEXT_PUBLIC_WORDPRESS_URL to a test WordPress site.
4. Run `npm run dev -w @faustwp/getting-started-example`
5. Go to your WordPress site, create a new post with a title & content and select "save as draft", then preview in a new tab.
6. Observe that the draft post is displaying the title & content.

You can also test the new Seed query by directly testing it out in GraphiQL.

### Query

```graphql
query GetSeedNode($id: ID = 0, $uri: String = "", $asPreview: Boolean = false) {
  ... on RootQuery @skip(if: $asPreview) {
    nodeByUri(uri: $uri) {
      __typename
      ...GetNode
    }
  }
  ... on RootQuery @include(if: $asPreview) {
    contentNode(id: $id, idType: DATABASE_ID, asPreview: true) {
      __typename
      ...GetNode
    }
  }
}

fragment GetNode on UniformResourceIdentifiable {
  __typename
  uri
  id
  ...DatabaseIdentifier
  ...ContentType
  ...User
  ...TermNode
  ...ContentNode
  ...MediaItem
  ...Page
}

fragment DatabaseIdentifier on DatabaseIdentifier {
  databaseId
}

fragment MediaItem on MediaItem {
  id
  mimeType
}

fragment ContentType on ContentType {
  name
  isFrontPage
  isPostsPage
}

fragment Page on Page {
  isFrontPage
  isPostsPage
}

fragment TermNode on TermNode {
  isTermNode
  slug
  taxonomyName
}

fragment ContentNode on ContentNode {
  isContentNode
  slug
  contentType {
    node {
      name
    }
  }
  template {
    templateName
  }
}

fragment User on User {
  name
  userId
  databaseId
}
```

### Variables

Example GraphQL variables for testing a draft post:

```json
{
  "asPreview": true,
  "id": 150
}
```

Example GraphQL variables for testing a published post:

```json
{
  "uri": "index.php?p=150"
}
```

## Screenshots

<img width="1879" alt="Screenshot 2023-06-05 at 3 37 39 PM" src="https://github.com/wpengine/faustjs/assets/6676674/89c9d984-8fd3-48a3-94bc-c7044534e03d">

<img width="1890" alt="Screenshot 2023-06-05 at 3 37 21 PM" src="https://github.com/wpengine/faustjs/assets/6676674/6e547703-f7c5-4550-86a1-a4574aacb3e6">
